### PR TITLE
DEV9: remove winpcap dependency on windows, TAP only

### PR DESCRIPTION
**Pull request by GovanifY**
This removes all the mandatory dependencies the plugin had and removes support for PCAP-based networking on windows.
PCAP based networking was basically MITMing all connections on an interface using an external mandatory driver. We did not needed any of the advanced filtering PCAP needed, nor sniffing on entire interfaces, so having a user setup a TAP interface linked to the network he wishes to bridge seems like the sanest option and is in line with what other emulators did to emulate broadband(eg dolphin).

This PR rewrote and switched to a TAP based networking system, which was in a primitive and non working state inside the plugin and was mostly renovated for the occasion.

### CODE REVIEW

**Review by RedDevilus — APPROVED** [review]
I tested several situations with Wireshark, while i havent tested community testers it works with network config and OPL loader. On master it doesn't even recognise it for some reason (maybe partially my fault) but this PR works even better than that.

**Review by refractionpcsx2 — APPROVED** [review]

File: plugins/dev9ghzdrk/pcap_io.cpp [hunk 1]
```diff
@@ -111,6 +121,8 @@ int GetMACAddress(char *adapter, mac_address* addr)
 
 int pcap_io_init(char *adapter)
 {
+#ifdef _WIN32
+#else
```
Comment (lightningterror) [inline plugins/dev9ghzdrk/pcap_io.cpp:125]: Tab these two so they match endif.

File: plugins/dev9ghzdrk/pcap_io.h [hunk 2]
```diff
@@ -14,14 +14,21 @@
...
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#pragma pack(push,1)
 
+
+#ifdef _WIN32
+#else
```
Comment (lightningterror) [inline plugins/dev9ghzdrk/pcap_io.h:29]: Forgot this one.

File: plugins/dev9ghzdrk/pcap_io.cpp [hunk 3]
```diff
@@ -183,6 +194,8 @@ int pcap_io_init(char *adapter)
 	pcap_io_running=1;
 	emu_printf("Ok.\n");
 	return 0;
```
Comment (lightningterror) [inline plugins/dev9ghzdrk/pcap_io.cpp:196]: You can remove this return.

File: plugins/dev9ghzdrk/pcap_io.cpp [hunk 4]
```diff
@@ -261,11 +280,14 @@ int pcap_io_get_dev_num()
 
 	pcap_freealldevs(alldevs);
 
+
```
Comment (lightningterror) [inline plugins/dev9ghzdrk/pcap_io.cpp:283]: I'd remove the extra lines, you can have empty one before return, same for the others.

**Review by lightningterror — APPROVED** [review]
Formatting can be fixed/improved during merge, I'm fine with it.

**Comment by CrunchBite** [comment]
Hey all, XLink Kai dev here. Is there any interest in adding support for our new "DDS" feature like I added to Dolphin in https://github.com/dolphin-emu/dolphin/pull/8853 ?
The gist: Send an ethernet frame to XLink encapsulated in a UDP packet and XLink will send it to the local LAN as well as any remote players (if playing online with others). No extra dependencies on wincap/npcap/libpcap on the PCSX2 end, just a UDP socket. Simple spec doc is here: https://www.teamxlink.co.uk/wiki/DDS_Protocol
If there is interest in supporting this please reach out.

**Comment by GovanifY** [comment]
@CrunchBite Hey, I've been dealing with DEV9 stuff lately.

Unfortunately I've been busy as of late, and the DEV9 plugin hasn't been merged into the core yet. I should be able to work back on PCSX2 next week, probably thursday.
Once this part is done there shouldn't be any blockers for us to proceed with your patch/port it over. Feel free to ping me in this issue, IRC or Discord, I'm `govanify`on freenode and `GovanifY#2869` on Discord.

**Comment by CrunchBite** [comment]
Sounds good! I'll be in touch.

**Pull request merged**